File: TUF/Number_System/BinaryToDecimal.ts
function binaryToDecimal(num: string): number {
    let n: number = num.length;
    let base: number = 1;
    let ans: number = 0;
    for (let i: number = n - 1; i >= 0; i--) {
        if (num.charAt(i) === '1') {
            ans += base;
        }
        base = base * 2;
    }
    return ans;
}

let num: string = '1011';
console.log(`${num} is decimal is: ${binaryToDecimal(num)}`);

export { };

File: TUF/Number_System/BinaryToDecimal.java
import java.util.Scanner;

public class BinaryToDecimal {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter a binary number to convert it into decimal: ");
        String binaryNum = input.next();
        input.close();
        System.out.println(binaryNum + " in Decimal: " + binaryToDecimal(binaryNum));
    }

    private static int binaryToDecimal(String num) {
        int n = num.length();
        int base = 1;
        int ans = 0;
        for (int i = n - 1; i >= 0; i--) {
            if (num.charAt(i) == '1') {
                ans += base;
            }
            base *= 2;
        }
        return ans;
    }
}